lib/screen/view_playlist/view_playlist.dart

```diff
@@ -0,0 +1,406 @@
+                                selectedTokens: playList?.tokenIDs,
+                              ),
+                              accountIdentities: accountIdentities,
+                            ),
+                          );
+                        },
+                        listener: (context, nftState) {},
+                      ),
+                    )
+                  ],
+                ),
+              ),
+              Positioned(
+                bottom: 0,
+                left: 0,
+                right: 0,
+                child: PlaylistControl(
+                  onPlayTap: () {
+                    final payload = ArtworkDetailPayload(
+                      accountIdentities,
+                      0,
+                      isPlaylist: true,
+                    );
+                    Navigator.of(context).pushNamed(
+                      AppRouter.artworkPreviewPage,
+                      arguments: payload,
+                    );
+                  },
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+}
+
+Widget assetsWidget(
+  BuildContext context,
+  List<AssetToken> tokens, {
+  Function(String tokenID, bool value)? onChanged,
+  List<String>? selectedTokens,
+  required List<ArtworkIdentity> accountIdentities,
+}) {
+  const int cellPerRowPhone = 3;
+  const int cellPerRowTablet = 6;
+  const double cellSpacing = 3.0;
+  int cellPerRow =
+      ResponsiveLayout.isMobile ? cellPerRowPhone : cellPerRowTablet;
+
+  final estimatedCellWidth = MediaQuery.of(context).size.width / cellPerRow -
+      cellSpacing * (cellPerRow - 1);
+  final cachedImageSize = (estimatedCellWidth * 3).ceil();
+
+  return GridView.builder(
+    gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+      crossAxisCount: cellPerRow,
+      crossAxisSpacing: cellSpacing,
+      mainAxisSpacing: cellSpacing,
+    ),
+    itemBuilder: (context, index) {
+      final asset = tokens[index];
+
+      return GestureDetector(
+        child: asset.pending == true && !asset.hasMetadata
+            ? PendingTokenWidget(
+                thumbnail: asset.galleryThumbnailURL,
+                tokenId: asset.tokenId,
+              )
+            : tokenGalleryWidget(
+                context,
+                asset,
+                cachedImageSize,
+              ),
+        onTap: () {
+          if (asset.pending == true && !asset.hasMetadata) return;
+
+          final index = tokens
+              .where((e) => e.pending != true || e.hasMetadata)
+              .toList()
+              .indexOf(asset);
+          final payload =
+              ArtworkDetailPayload(accountIdentities, index, isPlaylist: true);
+          Navigator.of(context)
+              .pushNamed(AppRouter.artworkPreviewPage, arguments: payload);
+        },
+      );
+    },
+    itemCount: tokens.length,
+  );
+}
+
+class ThubnailPlaylistItem extends StatefulWidget {
```

Comment:
I saw you defined it twice, one int `add_new_playlist.dart`, and one in here. why not create it just one?